src/v/cloud_storage/async_manifest_view.h

```diff
@@ -203,4 +221,272 @@ class materialized_manifest_cache {
     ss::semaphore_units<> _reserved;
```

Comment:
Can we use a named semaphore for the manifest cache? (comment in random place cuz github is silly)